src/Application/Persistence/RegularGivingMandateEventSubscriber.php

```diff
@@ -9,17 +9,26 @@
 use Doctrine\ORM\Events;
 use MatchBot\Application\Assertion;
 use MatchBot\Application\Messenger\MandateUpserted;
+use MatchBot\Domain\DonorAccount;
+use MatchBot\Domain\DonorAccountRepository;
 use MatchBot\Domain\RegularGivingMandate;
+use Psr\Container\ContainerInterface;
 use Stripe\Mandate;
 use Symfony\Component\Messenger\Envelope;
 use Symfony\Component\Messenger\RoutableMessageBus;
 
 class RegularGivingMandateEventSubscriber implements EventSubscriber
 {
+    private ?RoutableMessageBus $bus = null;
+    private ?DonorAccountRepository $donorRepository = null;
+
     /**
      * @psalm-suppress PossiblyUnusedMethod
+     *
+     * Accepts whole container to lazy-load properties on first use, and allow for a circular dependency - this depends
+     * on a repository which depends on the EntityManager which depends on this.
```

Comment:
The circular dependency isn't great of course. I also tried breaking it by adding the subscriber to the EM in app bootstrap or index php, after the container has returned the EM, but sequencing that with `IntegrationTests\CreateRegularGivingMandateTest::setUp` that runs the bootstrap and then put test doubles in the container is awkward.